### DISCUSSION TITLE
How to clear object with setValue properly?

### DISCUSSION THREAD

**Discussion by Plugo-SeongGonGo** [react-hook-form/react-hook-form#10542] [Q&A]
I have an object field that could be filled with an API response.
If the response has been changed, I want to clear the object, so I called setValue like `setValue('data', {})`.
But object values still remain if it is accessed by a nested path, like `const { field } = useController({ control, name: `data.${id}` })`.
Should I call `setValue` for every nested path, or is there a better way to clear an object?

Sandbox: [https://codesandbox.io/s/keen-alex-69cvl2](https://codesandbox.io/s/keen-alex-69cvl2)

**Comment by Muffinmann** [comment]
I think you could try: [reset](https://www.react-hook-form.com/api/useform/reset/)

**Reply by Plugo-SeongGonGo** [reply]
I guess `reset` depends on or updates `defaultValues`. Could `reset` be independent of `defaultValues` in this case?
